examples/enc-search-sample.rs

```diff
@@ -0,0 +1,421 @@
+// =============================================================================
+// This is a sample program in Rust that uses the ironoxide SDK to illustrate
+// various patterns for using the encrypted search features.
+//
+// To execute this program, you need to provide a file that contains a device
+// context. This file can be generated using the ironoxide-cli utility. You can
+// download this from https://github.com/IronCoreLabs/ironoxide-cli/releases;
+// if a compiled version for your platform or architecture is not available, you
+// can use cargo to build and install it. The program assumes the device
+// context file is named "example.devcon" in the current directory.
+//
+// Also note that this program creates two different groups - in order to avoid
+// errors when executing the program multiple times, it appends the current
+// time (as a time_t) when the program is started to the group IDs.
+//
+// Copyright (c) 2020  IronCore Labs, Inc.
+// =============================================================================
+
+use ironoxide::prelude::*;
+use lazy_static::lazy_static;
+use mut_static::MutStatic;
```

Comment:
Looks like this is being used for GLOBAL_STATE. We normally wouldn't recommend using this type of pattern in a real application, so I'm wondering if we can either do something else or perhaps explain why we're using it here (stand in for persistent storage?)

Reply:
You didn't think these comments were sufficient?
```
// Mock out some functions that would call the back-end service and return data in the real app
// We use some mutable global state to persist things that the service usually would.
```

Reply:
I did see those later on. They do help.